extensions/positron-assistant/src/models.ts

```diff
@@ -97,6 +105,25 @@ abstract class AILanguageModel implements positron.ai.LanguageModelChatProvider
 		this.provider = _config.provider;
 	}
 
+	async resolveConnection(token: vscode.CancellationToken): Promise<boolean> {
+		token.onCancellationRequested(() => {
+			return false;
+		});
+
+		try {
+			// send a test message to the model
+			await ai.generateText({
```

Comment:
This can take an unbounded amount of time during which there's no user feedback -- should we show a progress indicator?

Reply:
That's a good idea. This could probably be improved further with the dialog remaining open if an error occurred (as well as showing what went wrong).

![image](https://github.com/user-attachments/assets/fc3f27b4-1e5d-4ace-8b33-e4f95d91874b)